test/functional/p2p_invalid_messages.py

```diff
@@ -46,114 +48,34 @@ def set_test_params(self):
         self.setup_clean_chain = True
 
     def run_test(self):
-        """
-         . Test msg header
-        0. Send a bunch of large (4MB) messages of an unrecognized type. Check to see
-           that it isn't an effective DoS against the node.
-
-        1. Send an oversized (4MB+) message and check that we're disconnected.
-
-        2. Send a few messages with an incorrect data size in the header, ensure the
-           messages are ignored.
-        """
         self.test_magic_bytes()
         self.test_checksum()
         self.test_size()
         self.test_msgtype()
         self.test_large_inv()
+        self.test_resource_exhaustion()
 
-        node = self.nodes[0]
-        self.node = node
-        node.add_p2p_connection(P2PDataStore())
-        conn2 = node.add_p2p_connection(P2PDataStore())
-
-        msg_limit = 4 * 1000 * 1000  # 4MB, per MAX_PROTOCOL_MESSAGE_LENGTH
-        valid_data_limit = msg_limit - 5  # Account for the 4-byte length prefix
-
-        #
-        # 0.
-        #
-        # Send as large a message as is valid, ensure we aren't disconnected but
-        # also can't exhaust resources.
-        #
-        msg_at_size = msg_unrecognized(str_data="b" * valid_data_limit)
-        assert len(msg_at_size.serialize()) == msg_limit
+    def test_resource_exhaustion(self):
+        conn = self.nodes[0].add_p2p_connection(P2PDataStore())
+        msg_at_size = msg_unrecognized(str_data="b" * VALID_DATA_LIMIT)
+        assert len(msg_at_size.serialize()) == MSG_LIMIT
 
         self.log.info("Sending a bunch of large, junk messages to test memory exhaustion. May take a bit...")
 
         # Run a bunch of times to test for memory exhaustion.
         for _ in range(80):
-            node.p2p.send_message(msg_at_size)
+            self.nodes[0].p2p.send_message(msg_at_size)
```

Comment:
in commit 271949980d:

```suggestion
            conn.send_message(msg_at_size)
```

Reply:
Done